pkg/query-service/rules/thresholdRule.go

```diff
@@ -594,6 +587,10 @@ func (r *ThresholdRule) Eval(ctx context.Context, ts time.Time, queriers *Querie
 			annotations = append(annotations, labels.Label{Name: a.Name, Value: expand(a.Value)})
 		}
 
+		if r.isTest {
+			annotations = append(annotations, labels.Label{Name: AlertDescriptionLabel, Value: fmt.Sprintf("The rule threshold is set to %f, and the observed metric value is %f", r.targetVal(), smpl.V)})
+		}
```

Comment:
Can you add this at the time of rule creation?

Reply:
Ah I see this requires a value but don't find this random `r.isTest` checks around good.

Reply:
yes. this entire function Eval() is borrowed from prom and can probably be more modular. I can pick it up in next cycle. 
Do you have any suggestions to change this here?  I could move annotations for test in a separate function ?